applications/StructuralMechanicsApplication/custom_utilities/static_condensation_utility.cpp

```diff
@@ -0,0 +1,240 @@
+                                                                << " is singular !" << std::endl;
+            
+            //2.) K_cond = K11 - K12*inv(K22)*K21 
+            K_temp = prod(K_temp,sub_matrices[2]);
+            K_temp = prod(sub_matrices[1],K_temp);
+            K_temp = sub_matrices[0]-K_temp;
+
+            //3.) Fill rLeftHandSide to maintain same matrix size
+            const std::vector<int> remaining_dof_list = CreateRemainingDofList(rTheElement, rDofList);
+            rLeftHandSideMatrix.clear();
+
+            SizeType dofA = 0;
+            SizeType dofB = 0;
+            for (SizeType i=0; i<num_dofs_remaining; ++i)
+            {
+                dofA = remaining_dof_list[i];
+                for (SizeType j=0; j<num_dofs_remaining; ++j)
+                {
+                    dofB = remaining_dof_list[j];
+                    rLeftHandSideMatrix(dofA,dofB) = K_temp(i,j);
+                }
+            }
+            KRATOS_CATCH("")
+        }
+
+        std::vector<MatrixType> CalculateSchurComplements(
+            ElementType& rTheElement,
+			const MatrixType& rLeftHandSideMatrix,
+			const std::vector<int> & rDofList)
+        {
+            KRATOS_TRY;
+            // K11(0) K12(1)	
+            // K21(2) K22(3)		K22->dofs to be cond.
+            // rDofList -> List of dofs to be condensed
+            const std::vector<int> remaining_dof_list = CreateRemainingDofList(rTheElement, rDofList);
+            const SizeType num_dofs_condensed = rDofList.size();
+            const SizeType num_dofs_remaining = GetNumDofsElement(rTheElement)-num_dofs_condensed;
+
+            KRATOS_ERROR_IF(num_dofs_remaining != remaining_dof_list.size()) << "unequal remaining dof size" << std::endl;
+
+            std::vector<MatrixType> sub_matrices(4);
+            sub_matrices[0] = ZeroMatrix(num_dofs_remaining, num_dofs_remaining);
+            sub_matrices[1] = ZeroMatrix(num_dofs_remaining, num_dofs_condensed);
+            sub_matrices[2] = ZeroMatrix(num_dofs_condensed, num_dofs_remaining);
+            sub_matrices[3] = ZeroMatrix(num_dofs_condensed, num_dofs_condensed);
+
+            FillSchurComplements(sub_matrices[0], rLeftHandSideMatrix, remaining_dof_list, 
+                                 remaining_dof_list, num_dofs_remaining, num_dofs_remaining);
+
+            FillSchurComplements(sub_matrices[1], rLeftHandSideMatrix, remaining_dof_list, 
+                                 rDofList, num_dofs_remaining, num_dofs_condensed);
+
+            FillSchurComplements(sub_matrices[2], rLeftHandSideMatrix, rDofList, 
+                                 remaining_dof_list, num_dofs_condensed, num_dofs_remaining);
+
+            FillSchurComplements(sub_matrices[3], rLeftHandSideMatrix, rDofList, 
+                                 rDofList, num_dofs_condensed, num_dofs_condensed);
+
+            return sub_matrices;
+            KRATOS_CATCH("") 
+        }
+
+		std::vector<int> CreateRemainingDofList(
```

Comment:
Tabulatiosn and spaces, be aware

Reply:
that is strange. In my editor everything is fine but here it looks like that....